src/safeposix/syscalls/net_calls.rs

```diff
@@ -667,131 +694,113 @@ impl Cage {
-                                let domain = sockfdobj.domain;
+                            let mut newaddr = sockfdobj.localaddr.clone().unwrap();
+                            mutmetadata = NET_METADATA.write().unwrap();
+                            let mut newport = match mutmetadata._reserve_localport(newaddr.addr(), 0, sockfdobj.protocol, sockfdobj.domain, false) {
+                                Ok(portnum) => portnum,
+                                Err(errnum) => return errnum,
+                            };
+                            newaddr.set_port(newport);
 
-                                //THIS DEADLOCKS IF WE DON'T DROP THE FDTABLE HERE...
-                                //we need to figure out a way to make sure that the loop doesn't break when these are dropped, though
-                                //if we drop everything before getting into the loop, we need to make sure that the fd is a sock everytime
-                                //and probably that it hasn't changed in the time that it took to drop the lock to the table and regain it...
+                            let newipaddr = newaddr.addr().clone();
+                            newsockobj.localaddr = Some(newaddr);
+                            newsockobj.remoteaddr = Some(remote_addr.clone());
 
-                                let id = newsockobj.socketobjectid.clone();
-                                drop(sockfdobj);
-                                drop(wrappedfd);
-                                drop(filedesc_enum);
+                            //create socket object for new connected socket
+                            newsockobj.socketobjectid = match mutmetadata.insert_into_socketobjecttable(acceptedsock) {
+                                Ok(id) => Some(id),
+                                Err(errnum) => {
+                                    mutmetadata.listening_port_set.remove(&mux_port(newipaddr.clone(), newport, sockfdobj.domain, TCPPORT));
+                                    return errnum;
+                                }
+                            };
 
-                                //socket inserter code
-                                let socket_result = {
-                                    let wrappedsock = interface::RustRfc::new(interface::RustLock::new(Socket(newsockobj)));
+                            *addr = remote_addr; //populate addr with what address it connected to
+                            let domain = sockfdobj.domain;
 
-                                    let newfd = if let Some(fd) = self.get_next_fd(None, Some(&fdtable)) {
-                                        fd
-                                    } else {
-                                        return syscall_error(Errno::ENFILE, "accept", "no available file descriptor number could be found");
-                                    };
+                            //THIS DEADLOCKS IF WE DON'T DROP THE FDTABLE HERE...
```

Comment:
You also left this deadlock note, but I don't see the explicit drop? 

You do use drop explicitly a bunch in this PR, which I thought is sort of frowned upon in Rust?

Reply:
Drop does make locks no longer last the full scope. The note is leftover from when we did manual blocking so I will remove it.

Reply:
Maybe we can figure out more of best practices and better ways to do things when we do the merge into main?

Reply:
Sure. Just looked into it, I was thinking of something else ie calling the underlying drop mechanism. Calling drop like this seems fine.